### DISCUSSION TITLE
How to find and use OpenGL?

### DISCUSSION THREAD

**Discussion by ghost** [JulNadeauCA/BSDBuild#12] [General]
BSDBuild has scripts for `gle`, `glx`, and `glu`, but not plain `opengl` (that defines `HAVE_OPENGL`). Please correct me if I'm wrong.

Update: I'm wrong. The script for `opengl` is here:

https://github.com/JulNadeauCA/BSDBuild/blob/master/BSDBuild/opengl.pm

**Comment by JulNadeauCA** [comment]
The `opengl` test should define `HAVE_OPENGL`, `OPENGL_CFLAGS` and `OPENGL_LIBS` if libGL works.
`HAVE_GLU`, `HAVE_GLX`, `HAVE_GLEXT` and `HAVE_WGL` are defined if those interfaces are available.
Maybe it would be a good idea for the `opengl` module to also check for EGL and GLES3? I don't have those on my system to test.

Tests for non-core OpenGL related libraries are available as separate modules, those include `gle`, `glew` and `glu`.
